a) Determine el modelo de regresión lineal que le permita entender la relación entre la variable PIB_per_cápita y las variables regresoras Mortalidad_Infantil, Tasa_Alfabetismo_Femenino y Tasa_fecundidad.<br>
b) ¿Qué puede concluir respecto de los estadísticos globales del modelo?<br>
c)¿Qué puede concluir respecto de los estadísticos individuales del modelo?

In [ ]:
# Cargamos las librerias
import pandas as pd
import numpy as np
import scipy.stats as ss
import statsmodels.formula.api as smf

In [ ]:
#importamos el csv como nuestra base de datos
bdd = pd.read_csv('solemne_forma_1.csv', sep=';', decimal=",")
bdd.head()

Mortalidad_Infantil  Tasa_Alfabetismo_Femenino  PIB_per_cápita  \
0                  128                         37            1870   
1                  204                         22             130   
2                  202                         16             310   
3                  197                         65             570   
4                   96                         76            2050   

   Tasa_fecundidad  
0             6.66  
1             6.15  
2             7.00  
3             7.25  
4             3.81

In [ ]:
#verificamos que este bien formateado
bdd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64 entries, 0 to 63
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Mortalidad_Infantil        64 non-null     int64  
 1   Tasa_Alfabetismo_Femenino  64 non-null     int64  
 2   PIB_per_cápita             64 non-null     int64  
 3   Tasa_fecundidad            64 non-null     float64
dtypes: float64(1), int64(3)
memory usage: 2.1 KB


In [ ]:
# Usamos el método de los mínimos cuadrados (ols) para ajustar la regresión lineal
mod1 = smf.ols('PIB_per_cápita ~ Mortalidad_Infantil+	Tasa_Alfabetismo_Femenino+Tasa_fecundidad', bdd).fit()
# Desplegamos resultados del modelo
mod1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:         PIB_per_cápita   R-squared:                       0.164
Model:                            OLS   Adj. R-squared:                  0.122
Method:                 Least Squares   F-statistic:                     3.931
Date:                Tue, 03 Jan 2023   Prob (F-statistic):             0.0125
Time:                        00:54:52   Log-Likelihood:                -590.85
No. Observations:                  64   AIC:                             1190.
Df Residuals:                      60   BIC:                             1198.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=============================================================================================
                                coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------
Intercept                  3825.5536   2489.125      1.537      0.130   -1153.439    8804.546
Mortalidad_Infantil         -18.6792      7.284     -2.565      0.013     -33.249      -4.110
Tasa_Alfabetismo_Femenino   -11.4455     21.377     -0.535      0.594     -54.206      31.315
Tasa_fecundidad             139.0045    279.714      0.497      0.621    -420.507     698.516
==============================================================================
Omnibus:                      103.449   Durbin-Watson:                   2.335
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             2524.104
Skew:                           4.931   Prob(JB):                         0.00
Kurtosis:                      32.142   Cond. No.                     1.28e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.28e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

a) Utilizando los datos desplegados por el summary, el modelo de regresión lineal es:<br>
$pib\_per\_cápita=3825.5536−18.6792∗Mortalidad\_Infantil−11.4455∗Tasa\_Alfabetismo\_Femenino+139.0045∗Tasa\_fecundidad$

b) A partir de los valores de F estadistico, R2 y R2 Adj podemos concluir que el modelo es capaz de explicar en un 12.2% la varianza.
En cuanto a F, es significativo ($F_{estadistico} < p_{valor}$). <br>
Teniendo que:<br>
$H_0 = los\ coeficientes\ de\ regresiones\ son\ iguales\ a\ cero$<br>
$H_1 = Al\ menos\ un\ coeficiente\ debe\ ser\ son\ distinto \ de\ cero$<br>
En vista de los datos obtenidos, Se rechaza la hipótesis nula.


c)
Teniendo que la hipotesis nula para las variables es :<br>
$H_0 = los\ coeficientes\ de\ regresion\ lineal\ es\ igual\ a\ cero$<hr>
Los coeficientes asociados a la variable Mortalidad_Infantil mantienen un t estadísticamente significativo ($ t_{estadistico} < p_{valor}$), por lo que se rechaza la hipótesis nula para estas variables.<br>En los casos de  Tasa_Alfabetismo_Femenino y Tasa_fecundidad se acepta la hipótesis nula por lo que su coeficiente de regresión es igual a cero.
